### PR TITLE
[Eventonica React] - merge existing content with additional porting instructions

### PR DESCRIPTION
**Pull request by vegetabill**
Originally opened as #1472 but I messed up that branch in Git 😬 

- include specific walkthrough of porting JS DOM code => React
- Link to other new content since originally created
- Keep all server / API hints

### CODE REVIEW

File: projects/eventonica/eventonica-part7-react.md [hunk 1]
```diff
@@ -72,10 +70,145 @@ Here's an example of what `package.json` might look like now:
...
+
+#### JavaScript
+
+You should be able to move any JavaScript files into the `public` folder and use them as-is. At this point your app should work but not really using React in any way. Make sure the basics work before proceeding.
+
+### Create Your First Component
+
+Before this stage, ensure you have a commit in place with the working app.
+
+Here is example code that might be similar to the way you displayed a list of users in your app. Depending on how you built Part 2, your code might not be in a function like this but hopefully you can translate the concept to your own code.
```
Comment (vegetabill) [inline projects/eventonica/eventonica-part7-react.md:89]: Fixed typos from #1472

File: projects/eventonica/eventonica-part7-react.md [hunk 2]
```diff
@@ -72,10 +70,145 @@ Here's an example of what `package.json` might look like now:
...
+        <input id="delete-user-action" type="submit" value="Delete User" />
+      </form>
+    );
+  }
+}
+```
+
+### Re-rendering the UI in Response to Changes
+
+Now we want the UI to re-render when we delete the user. To do this we will have to call `this.setState` with updated data to ensure that React will re-render. Use the previous lessons you went through to figure out how to do this.
```
Comment (vegetabill) [inline projects/eventonica/eventonica-part7-react.md:209]: Please review this new wording, @alodahl . The last wording was confusing and came from me having mixed results teaching immutable code patterns as the participants might see in a real codebase vs. modifying and objects in state and then just calling `this.setState` with the whole object again.

File: projects/eventonica/eventonica-part7-react.md [hunk 3]
```diff
@@ -72,10 +70,145 @@ Here's an example of what `package.json` might look like now:
...
+    displayEvents(); // refresh the DOM
+  });
+```
+
+In React we would setup the event handler like this:
+
+```jsx
+// src/UserDeleteForm.js
+class UserDeleteForm extends React.Component {
+  state = { idToDelete: null };
```
Comment (alodahl) [inline projects/eventonica/eventonica-part7-react.md:182]: Thanks for doing this!  Again, I'm not sure this is the way to initiate state... I think you need a constructor and everything.

**Review by alodahl — CHANGES_REQUESTED** [review]

File: projects/eventonica/eventonica-part7-react.md (hunk 3 above)
Comment (alodahl) [inline projects/eventonica/eventonica-part7-react.md:182]:
```suggestion
  constructor(props) {
  super(props);
  this.state = { idToDelete: null };
}
```

File: projects/eventonica/eventonica-part7-react.md (hunk 2 above)
Comment (alodahl) [reply projects/eventonica/eventonica-part7-react.md:209]: One thing that's really important for React that you don't have to care about with jQuery is that the user ID list should be in the state of the parent component, meaning that the onClick will need to be a callback prop, and you probably want the callback function to live in the parent to make it clear.  If you think they've learned enough to accomplish this, maybe you can just leave a note reminding them of these things, but let them write the code themselves?

**Pull request merged**